### PR TITLE
Removed assumptions in test_solveset.py

### LINKED ISSUE #18437: test_solveset.py: unintended(?) assumptions in lots of tests
Lots of tests in https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_solveset.py use "predefined" variables `x`, `y`, etc.
This would be fine, except those have all been created with `real=True` assumption!
https://github.com/sympy/sympy/blob/8a7dc8ef786cccbd7dc0fe0b0e6beec4dd1f7b49/sympy/solvers/tests/test_solveset.py#L47-L55

I suppose this is unintended and falsifies a *lot* of the tests.

So to resolve this issue, we should
- remove the assumptions (and possible re-add them where they make sense, e.g. in reals-only tests)
- see what tests break
- fix them, file bugs, etc.

Also of note is that `solveset` is documented to be independent of assumptions: https://docs.sympy.org/1.5.1/modules/solvers/solveset.html#what-is-this-domain-argument-about
So at least the choice of internal solvers does not depend on those assumptions, but some simplifications will or will not happen based on them.

The upside is that this can be done almost test by test.

### PR DESCRIPTION
**Pull request by risubaba**
#### References to other Issues or PRs

Fixes #18437 

#### Brief description of what is fixed or changed
Variables were set globally to be Real , that has been changed and the assumption has been added only when required

#### Other comments
This could probably be fixed differently by adding the declaration of symbol within every test but this works aswell.
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by gschintgen** [comment]
I only had a quick look at the commit, but I think there's a misunderstanding: The `real=True` assumption should **only** be added back for those specific tests where they are **needed** for the test to succeed. In general it should not be necessary to add the assumption simply because the test calls `solveset_real`.

In order to keep the history clean and since there are no review comments for specific changed lines it may be best to force-push a new commit to this branch that replaces the current one.

Also, but this may be a matter of personal taste, I find the variable name `xx` rather unconventional. It could just as well be `Symbol('r', real=True)` or even `x` if it's only a short test function and all the tests need the assumption.

**Comment by gschintgen** [comment]
According to the guidelines, this should not have a release notes entry. (Tests are not relevant for the end user.)
See https://github.com/sympy/sympy/wiki/Writing-Release-Notes#what-makes-a-good-release-notes-entry

**Comment by risubaba** [comment]
@gschintgen I removed the `real=True` assumption and added it only when the tests were failing. This is what you are also saying right?

**Comment by oscarbenjamin** [comment]
Maybe the best way is to remove the real=True from the variables at the top and then underneath add
```
xr = Symbol('xr', real=True)
...
```
Then where there is a test that is failing change e.g. `x` to `xr` but *only* in the exact assert that is failing. You should not change any assert that still works after removing the real=True assumption.

**Comment by risubaba** [comment]
@oscarbenjamin Thanks for the advice. I'll do this and commit in some time.

**Comment by gschintgen** [comment]
> @gschintgen I removed the `real=True` assumption and added it only when the tests were failing

As pointed out by @oscarbenjamin this should only be done for single asserts. E.g. this:
```
    assert solveset_real(xx, 1) == S.EmptySet
```
does not need the assumption to succeed.

**Comment by risubaba** [comment]
I will update the PR by tomorrow.

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -44,15 +44,15 @@
...
-n = Symbol('n', real=True)
+a = Symbol('a')
+b = Symbol('b')
+c = Symbol('c')
+x = Symbol('x')
+y = Symbol('y')
+z = Symbol('z')
+q = Symbol('q')
+m = Symbol('m')
+n = Symbol('n')
```
Comment (czgdp1807) [inline sympy/solvers/tests/test_solveset.py]: `a, b, c, x, y, z, q, m, n = symbols('a b c x y z q m n')`

**Comment by czgdp1807** [comment]
@risubaba Any updates?

**Comment by risubaba** [comment]
@czgdp1807 Really sorry for the late reply. I have been busy with some other work and won't be able to finish this.

**Comment by czgdp1807** [comment]
Looks like someone can continue this. Re-opening and attaching a `Please take over` label. Thanks for the update @risubaba

File: sympy/solvers/tests/test_solveset.py [hunk 2]
```diff
@@ -1138,8 +1126,7 @@ def test_abs_invert_solvify():
 
 
 def test_linear_eq_to_matrix():
-    x, y, z = symbols('x, y, z')
-    a, b, c, d, e, f, g, h, i, j, k, l = symbols('a:l')
+    from sympy.abc import d, e, f, g, h, i, j, k, l
```
Comment (smichr) [inline sympy/solvers/tests/test_solveset.py]: use d-l at top

**Comment by czgdp1807** [comment]
Thank you @smichr for completing this.

**Pull request merged**